smartsim/_core/entrypoints/file_operations.py

```diff
@@ -0,0 +1,272 @@
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import base64
+import collections
+import os
+import pickle
+import re
+import shutil
+from ast import literal_eval
+from distutils import dir_util  # pylint: disable=deprecated-module
+
+
+def _check_path(file_path: str) -> str:
+    """Given a user provided path-like str, find the actual path to
+        the directory or file and create a full path.
+
+    :param file_path: path to a specific file or directory
+    :raises FileNotFoundError: if file or directory does not exist
+    :return: full path to file or directory
+    """
+    full_path = os.path.abspath(file_path)
+    if os.path.isfile(full_path):
+        return full_path
+    if os.path.isdir(full_path):
+        return full_path
+    raise FileNotFoundError(f"File or Directory {file_path} not found")
+
+
+def move(parsed_args: argparse.Namespace) -> None:
+    """Move a file
+
+    Sample usage:
+        python _core/entrypoints/file_operations.py move
+        /absolute/file/src/path /absolute/file/dest/path
+
+        source path: Path to a source file to be copied
+        dest path: Path to a file to copy the contents from the source file into
+    """
+    _check_path(parsed_args.source)
+    _check_path(parsed_args.dest)
+    shutil.move(parsed_args.source, parsed_args.dest)
+
+
+def remove(parsed_args: argparse.Namespace) -> None:
+    """Write a python script that removes a file when executed.
+
+    Sample usage:
+        python _core/entrypoints/file_operations.py remove
+        /absolute/file/path
+
+        file path: Path to the file to be deleted
+    """
+    _check_path(parsed_args.to_remove)
+    os.remove(parsed_args.to_remove)
+
+
+def copy(parsed_args: argparse.Namespace) -> None:
+    """
+    Write a python script to copy the entity files and directories attached
+    to this entity into an entity directory
+
+    Sample usage:
+        python _core/entrypoints/file_operations.py copy
+        /absolute/file/src/path /absolute/file/dest/path
+
+        source path: Path to directory, or path to file to copy into an entity directory
+        dest path: Path to destination directory or path to destination file to copy
+    """
+    _check_path(parsed_args.source)
+    _check_path(parsed_args.dest)
```

Comment:
Same question as before.  If the destination file/path doesn't exist yet, won't this raise an error?  Is that the intended behavior?  

I'm wondering if ``_check_path`` is really necessary... Are we better off just catching the errors raise by the python methods and reraising as a SmartSim error if we wanted to a cleaner error class?

Reply:
I can remove these